nap/utils.py

```diff
@@ -60,19 +51,21 @@ def flatten_params(k, vs):
 
         # since we can have more than one value for a single key, we use a
         # tuple of two tuples instead of a dictionary
-        params_tuple = tuple(itertools.chain(*flat_params))
-        param_string = urllib.parse.urlencode(params_tuple)
+        params_tuple = tuple(sorted(itertools.chain(*flat_params), key=lambda key_value: key_value[0]))
```

Comment:
I said _add_ the value to be part of the key function, meaning in addition to the key. That way multiple values for the same param name would appear in the same order in the resulting url string regardless of their original order.

Something like this:
`params_tuple = tuple(sorted(itertools.chain(*flat_params), key=itemgetter(0,1)))`